src/eduid_userdb/security/db.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 NORDUnet A/S
+# All rights reserved.
+#
+#   Redistribution and use in source and binary forms, with or
+#   without modification, are permitted provided that the following
+#   conditions are met:
+#
+#     1. Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#     2. Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided
+#        with the distribution.
+#     3. Neither the name of the NORDUnet nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+from __future__ import absolute_import
+
+from eduid_userdb.userdb import UserDB
+from eduid_userdb.security import SecurityUser
+
+import logging
+logger = logging.getLogger(__name__)
+
+__author__ = 'lundberg'
+
+
+class SecurityUserDB(UserDB):
+
+    UserClass = SecurityUser
+
+    def __init__(self, db_uri, db_name='eduid_security', collection='profiles'):
+        super(SecurityUserDB, self).__init__(db_uri, db_name, collection=collection)
+
+    def save(self, user, check_sync=True, old_format=False):
+        if not isinstance(user, SecurityUser):
+            user = SecurityUser(data=user.to_dict())
```

Comment:
is permitting other User classes than SecurityUser here useful or just a great way to shoot myself in the foot?

Reply:
Hmm, at least this if statement should be moved to UserDB.

Should we raise an exception if the user does not match the dbs UserClass or should we try to convert it... I don't know.